### ISSUE TITLE
new rule: `no-nested-style-tag`

### ISSUE THREAD

**Issue by baseballyama** [sveltejs/eslint-plugin-svelte#1186]
### Motivation

This rule prohibits the use of nested <style> tags to prevent unintended style application.

https://svelte.dev/docs/svelte/nested-style-elements

### Description

Nested <style> tags are not scoped, which may lead to unintended styles being applied.

### Examples

```svelte
<script>
</script>

<!-- ✓ GOOD -->
<style>...</style>

<!-- ✗ BAD -->
<div>
  <style>...</style>
</div>

{@snippet}
  <style>...</style>
{/snippet}
```
